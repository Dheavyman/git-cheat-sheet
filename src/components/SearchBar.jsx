import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { handleSearch } from '../actions/cheats';

const propTypes = {
  actions: PropTypes.shape({
    handleSearch: PropTypes.func.isRequired
  }).isRequired
};

/**
 * Search bar component
 *
 * @returns {object} React element
 */
export const SearchBar = ({ actions: { handleSearch } }) => (
  <div className="row search-bar">
    <form className="col s6 offset-s3">
      <div className="row">
        <div className="input-field col s12">
          <i className="material-icons prefix">search</i>
          <input
            id="search"
            type="text"
            className="center-align validate"
            onChange={event => event.target.value.length > 1
              ? handleSearch(event.target.value)
              : handleSearch('')}
          />
          <label htmlFor="search">THE AWESOME GIT CHEAT SHEET</label>
        </div>
      </div>
    </form>
  </div>
);

SearchBar.propTypes = propTypes;

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ handleSearch }, dispatch)
});

export default connect(null, mapDispatchToProps)(SearchBar);
