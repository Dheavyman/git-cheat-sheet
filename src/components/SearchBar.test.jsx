import React from 'react';
import TestRenderer from 'react-test-renderer';

import { SearchBar } from './SearchBar';

describe('SearchBar component', () => {
  const props = {
    actions: {
      handleSearch: jest.fn()
    }
  };
  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<SearchBar {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
