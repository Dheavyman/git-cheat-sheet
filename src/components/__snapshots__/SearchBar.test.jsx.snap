// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should render component successfully 1`] = `
<div
  className="row search-bar"
>
  <form
    className="col s6 offset-s3"
  >
    <div
      className="row"
    >
      <div
        className="input-field col s12"
      >
        <i
          className="material-icons prefix"
        >
          search
        </i>
        <input
          className="center-align validate"
          id="search"
          onChange={[Function]}
          type="text"
        />
        <label
          htmlFor="search"
        >
          THE AWESOME GIT CHEAT SHEET
        </label>
      </div>
    </div>
  </form>
</div>
`;
